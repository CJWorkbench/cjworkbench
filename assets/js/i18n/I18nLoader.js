import React from 'react'
import { I18nProvider } from '@lingui/react'
import fetchCatalog from './catalogs'

export class I18nLoader extends React.Component {
  state = {
    catalogs: {},
  }
  
  loadCatalog = (language) => {
      this.setState(state => ({
          catalogs: {
            ...state.catalogs,
            [language]: fetchCatalog(language)
          }
      }))
  }
  
  componentDidMount() {
    this.loadCatalog(this.props.language)
  }

  shouldComponentUpdate(nextProps, nextState) {
    const { language } = nextProps
    const { catalogs } = nextState

    if (language !== this.props.language && !catalogs[language]) {
      this.loadCatalog(language)
      return false
    }

    return true
  }

  render () {
    const { children, language } = this.props
    const { catalogs } = this.state

    // Skip rendering when catalog isn't loaded.
    if (!catalogs[language]) {
        return null
    }

    return (
      <I18nProvider language={language} catalogs={catalogs}>
        {children}
      </I18nProvider>
    )
  }
}
