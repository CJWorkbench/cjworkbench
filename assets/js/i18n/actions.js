const SET_LOCALE = 'SET_LOCALE';

function reduceSetLocale(state, action){
  return {
    ...state,
    locale: action.locale
  };
}

export function setLocaleAction(locale){
  return {
    type: SET_LOCALE,
    locale: locale
  }
}

export const reducerFunctions = {
  [SET_LOCALE]: reduceSetLocale
}

export function localeReducer(state, action){
  if (action.type in reducerFunctions) {
    // Run a registered reducer
    return reducerFunctions[action.type](state, action)
  }
  
  return state;
}
