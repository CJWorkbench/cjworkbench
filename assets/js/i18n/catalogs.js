import enCatalog from '../../locale/en/messages.po'
import elCatalog from '../../locale/el/messages.po'

const catalogs = {
  en: enCatalog,
  el: elCatalog
}

export default function fetchCatalog (language) {
  return catalogs[language]
}
