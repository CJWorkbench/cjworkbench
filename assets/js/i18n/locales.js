import { t } from '@lingui/macro'

export const supportedLocales = {
  en: t('js.i18n.locales.en')`English`,
  el: t('js.i18n.locales.el')`Greek`
}

export const supportedLocaleIds = Object.keys(supportedLocales)

const defaultLocale = 'en'

export const currentLocale = window.i18nConfig && window.i18nConfig.locale && isSupported(window.i18nConfig.locale) ? window.i18nConfig.locale : defaultLocale

function isSupported (locale) {
  return Object.prototype.hasOwnProperty.call(supportedLocales, locale)
}
