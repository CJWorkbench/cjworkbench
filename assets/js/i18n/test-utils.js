import { I18nProvider } from '@lingui/react'
import React from 'react'
import { shape, object } from 'prop-types'
import { mount, shallow } from 'enzyme'

/*
 * Below lie utils for testing with i18n.
 * See https://lingui.js.org/guides/testing.html
 */
// Create the I18nProvider to retrieve context for wrapping around.
const intlProvider = new I18nProvider({
  language: 'en',
  catalogs: {
    en: {}
  }
}, {})

const {
  linguiPublisher: {
    i18n: originalI18n
  }
} = intlProvider.getChildContext()

// You customize the i18n object here:
export const i18n = {
  ...originalI18n,
  // provide _ macro, for trying to unwrap default/id or just passing down the key
  _: key => {
    if (typeof key.defaults === 'string') return key.defaults
    if (typeof key.id === 'string') return key.id
    return key
  }
}

/**
 * When using Lingui `withI18n` on components, props.i18n is required.
 */
function nodeWithI18nProp (node) {
  return React.cloneElement(node, { i18n })
}

/*
 * Methods to use
 */
export function shallowWithI18n (node, { context } = {}) {
  return shallow(
    nodeWithI18nProp(node),
    {
      context: Object.assign({}, context, { i18n })
    }
  )
}

export function mountWithI18n (node, { context, childContextTypes } = {}) {
  const newContext = Object.assign({}, context, { linguiPublisher: { i18n } })
  /*
   * I18nProvider sets the linguiPublisher in the context for withI18n to get
   * the i18n object from.
   */
  const newChildContextTypes = Object.assign({},
    {
      linguiPublisher: shape({
        i18n: object.isRequired
      }).isRequired
    },
    childContextTypes
  )
  return mount(
    nodeWithI18nProp(node),
    {
      context: newContext,
      childContextTypes: newChildContextTypes
    }
  )
}
