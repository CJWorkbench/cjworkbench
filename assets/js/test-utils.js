/* globals jest */
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import promiseMiddleware from 'redux-promise-middleware'
import errorMiddleware from './error-middleware'
import { workflowReducer } from './workflow-reducer'

// Returns new mock function that returns given json. Used for mocking "get" API calls
export function jsonResponseMock (json) {
  return jest.fn().mockImplementation(() =>
    Promise.resolve(json)
  )
}

// Returns new mock function that gives an OK HTTP response. Use for mocking "set" API calls
export function okResponseMock () {
  return jsonResponseMock(null)
}

// Helper function to return a promise that resolves after all other promise mocks,
// even if they are chained like Promise.resolve().then(...).then(...)
// Technically: this is designed to resolve on the next macrotask
// https://stackoverflow.com/questions/25915634/difference-between-microtask-and-macrotask-within-an-event-loop-context
export function tick () {
  return new Promise(resolve => {
    setTimeout(resolve, 0)
  })
}

export function sleep (ms) {
  return new Promise(resolve => {
    setTimeout(resolve, ms)
  })
}

/**
 * Return `[ notify, promise ]`; `promise` completes when `notify` is called.
 *
 * Usage:
 *
 *     const [ notify, condition ] = createConditionVariable
 *     const done = condition.then(do_something)
 *     await Promise.all([sleep(0.1).then(notify), done])
 */
export function createConditionVariable () {
  let notify
  const promise = new Promise(resolve => { notify = resolve })
  return [notify, promise]
}

/**
 * Mock Redux store with in-production reducers.
 *
 * This is for integration-style tests: they test the reducer _and_ the caller
 * at the same time.
 *
 * Usage:
 *
 *     const api = { undo: jest.fn() }
 *     const store = mockStoreWithReducer({'workflow': ..., ...}, api)
 *     store.dispatch(Actions.undo())
 *     expect(api.undo).toHaveBeenCalled()
 *     expect(store.getState().workflow).toEqual(...)
 */
export function mockStore (initialState, api = {}) {
  const middlewares = [errorMiddleware(), promiseMiddleware, thunk.withExtraArgument(api)]
  const store = createStore(workflowReducer, initialState, applyMiddleware(...middlewares))
  return store
}
