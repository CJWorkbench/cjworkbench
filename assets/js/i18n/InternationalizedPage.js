import React from 'react'
import PropTypes from 'prop-types'
import { en, el } from 'make-plural/plurals'
import { i18n } from '@lingui/core'
import { I18nProvider } from '@lingui/react'
import { currentLocaleId } from './locales'
import Spinner from '../Spinner'

__webpack_public_path__ = window.STATIC_URL + 'bundles/' // eslint-disable-line

async function loadLocale (localeId) {
  const { messages } = await import(
    /* webpackChunkName: "[request]" */
    `@lingui/loader!../../locale/${localeId}/messages.po`
  )
  i18n.loadLocaleData({
    en: { plurals: en },
    el: { plurals: el }
  })
  i18n.load(localeId, messages)
  i18n.activate(localeId)
  return i18n
}

function useActivatedI18n (localeId) {
  const [i18n, setI18n] = React.useState(null)

  React.useEffect(() => {
    loadLocale(currentLocaleId).then(setI18n)
  }, [])

  return i18n
}

export default function InternationalizedPage (props) {
  const { children } = props
  const i18n = useActivatedI18n(currentLocaleId)

  if (i18n === null) {
    return <Spinner />
  }

  return (
    <I18nProvider i18n={i18n} forceRenderOnLocaleChange={false}>
      {children}
    </I18nProvider>
  )
}
InternationalizedPage.propTypes = {
  children: PropTypes.node.isRequired
}
