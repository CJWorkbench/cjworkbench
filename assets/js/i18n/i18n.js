import { setupI18n } from "@lingui/core"
import { currentLocale } from './locales'
import { fetchCatalog } from './catalogs'

export default const i18n = setupI18n({
   language: currentLocale,
   catalogs: {
       currentLocale: fetchCatalog(currentLocale)
   }
})
