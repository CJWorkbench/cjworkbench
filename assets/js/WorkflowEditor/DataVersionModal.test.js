/* globals afterEach, describe, expect, it, jest */
import React from 'react'
import { mountWithI18n } from '../i18n/test-utils'
import ConnectedDataVersionModal, { DataVersionModal, formatDateUTCForTesting } from './DataVersionModal'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'

describe('DataVersionModal', () => {
  // Make formatDate in DataVersionModal always print out UTC times
  formatDateUTCForTesting()

  const Versions = [
    { id: '2000', date: new Date(234567890), isSeen: false },
    { id: '1000', date: new Date(123456789), isSeen: true }
  ]

  let _wrapper = null

  afterEach(() => {
    if (_wrapper !== null) {
      _wrapper.unmount()
      _wrapper = null
    }
  })

  const wrapper = (extraProps) => {
    _wrapper = mountWithI18n(
      <DataVersionModal
        fetchWfModuleId={123}
        fetchWfModuleName='fetch'
        fetchVersions={Versions}
        selectedFetchVersionId='1000'
        wfModuleId={124}
        isAnonymous={false}
        notificationsEnabled={false}
        onClose={jest.fn()}
        onChangeFetchVersionId={jest.fn()}
        onChangeNotificationsEnabled={jest.fn()}
        {...extraProps}
      />
    )
    return _wrapper
  }

  it('matches snapshot', () => {
    expect(wrapper()).toMatchSnapshot()
  })

  it('displays versions', () => {
    const w = wrapper()
    expect(w.find('label.seen.selected time').text()).toEqual('Jan. 2, 1970 – 10:17 a.m.')
    expect(w.find('label.unseen time').text()).toEqual('Jan. 3, 1970 – 5:09 p.m.')
  })

  it('selects a version', () => {
    const w = wrapper()

    // Click new version; verify it's clicked
    w.find('label.unseen input').simulate('change', { target: { checked: true } })
    expect(w.find('label.unseen input').prop('checked')).toBe(true)

    // Click 'Load'
    w.find('button[name="load"]').simulate('click')
    expect(w.prop('onChangeFetchVersionId')).toHaveBeenCalledWith(123, '2000')
    expect(w.prop('onClose')).toHaveBeenCalled()
  })

  it('cancels with a close button', () => {
    const w = wrapper()

    // Click new version; verify it's clicked
    w.find('label.unseen input').simulate('change', { target: { checked: true } })

    w.find('button.close').simulate('click')
    expect(w.prop('onClose')).toHaveBeenCalled()
  })

  it('enables/disables notifications', () => {
    const w = wrapper()

    w.find('input[name="notifications-enabled"]').simulate('change', { target: { checked: true } })
    expect(w.prop('onChangeNotificationsEnabled')).toHaveBeenCalledWith(124, true)
    w.setProps({ notificationsEnabled: true }) // simulate onChangeNotificationsEnabled()
    expect(w.find('input[name="notifications-enabled"]').prop('checked')).toBe(true)

    w.find('input[name="notifications-enabled"]').simulate('change', { target: { checked: false } })
    expect(w.prop('onChangeNotificationsEnabled')).toHaveBeenCalledWith(124, true)
    w.setProps({ notificationsEnabled: false }) // simulate onChangeNotificationsEnabled()
    expect(w.find('input[name="notifications-enabled"]').prop('checked')).toBe(false)
  })

  it('omits notifications settings when isAnonymous', () => {
    const w = wrapper({ isAnonymous: true })
    expect(w.find('form.notifications').length).toBe(0)
  })

  describe('mapStateToProps', () => {
    // Assume this modal is never shown if there is no fetch module
    const IdealState = {
      workflow: {
        is_anonymous: false,
        tab_slugs: ['tab-11'],
        selected_tab_position: 0
      },
      tabs: {
        'tab-11': { wf_module_ids: [123, 124] }
      },
      modules: {
        fetch: { name: 'Fetch Stuff', loads_data: true },
        filter: { name: 'Filter Stuff', loads_data: false }
      },
      wfModules: {
        123: {
          id: 123,
          notifications: true,
          module: 'fetch',
          versions: {
            versions: [
              ['2018-06-22T20:09:41.649Z', true],
              ['2018-06-23T20:09:41.649Z', false]
            ],
            selected: '2018-06-22T20:09:41.649Z'
          }
        },
        124: {
          id: 124,
          module: 'filter',
          name: 'Filter Stuff',
          notifications: true
        }
      }
    }

    const connectedWrapper = (state) => {
      const store = configureMockStore([])(state)
      _wrapper = mountWithI18n(
        <Provider store={store}>
          <ConnectedDataVersionModal
            wfModuleId={124}
            onClose={jest.fn()}
          />
        </Provider>
      )
      return _wrapper
    }

    it('should find notificationsEnabled', () => {
      const w = connectedWrapper(IdealState)
      expect(w.find('input[name="notifications-enabled"]').prop('checked')).toBe(true)
    })

    // it('should set fetchModuleName', () => {
    //  const w = connectedWrapper(IdealState)
    //  // expect(w.find('p.introduction').text()).toMatch(/“Fetch Stuff”/) No introduction test for now (Pierre)
    // })

    it('should set fetchVersions', () => {
      const w = connectedWrapper(IdealState)

      expect(w.find('label.selected time[time="2018-06-22T20:09:41.649Z"]').length).toBe(1)
      expect(w.find('label.unseen time[time="2018-06-23T20:09:41.649Z"]').length).toBe(1)
    })
  })
})
