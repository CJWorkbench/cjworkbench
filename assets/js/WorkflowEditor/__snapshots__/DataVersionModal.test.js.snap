// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataVersionModal matches snapshot 1`] = `
<DataVersionModal
  fetchVersions={
    Array [
      Object {
        "date": 1970-01-03T17:09:27.890Z,
        "id": "2000",
        "isSeen": false,
      },
      Object {
        "date": 1970-01-02T10:17:36.789Z,
        "id": "1000",
        "isSeen": true,
      },
    ]
  }
  fetchWfModuleId={123}
  fetchWfModuleName="fetch"
  isAnonymous={false}
  notificationsEnabled={false}
  onChangeFetchVersionId={[MockFunction]}
  onChangeNotificationsEnabled={[MockFunction]}
  onClose={[MockFunction]}
  selectedFetchVersionId="1000"
  wfModuleId={124}
>
  <Modal
    className="data-versions-modal"
    fade={false}
    isOpen={true}
    toggle={[MockFunction]}
  >
    <OpenModal
      className="data-versions-modal"
      fade={false}
      toggle={[MockFunction]}
    >
      <Portal
        containerInfo={
          <body
            class="modal-open"
          >
            <div
              class="modal-backdrop"
            />
            <div
              class="modal data-versions-modal"
              role="dialog"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      Data Versions
                    </h5>
                    <button
                      aria-label="Close"
                      class="close"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <form>
                      <ol>
                        <li>
                          <label
                            class="unseen"
                          >
                            <input
                              name="data-version"
                              type="radio"
                              value="2000"
                            />
                            <time
                              time="1970-01-03T17:09:27.890Z"
                            >
                              Jan. 3, 1970 – 5:09 p.m.
                            </time>
                          </label>
                        </li>
                        <li>
                          <label
                            class="seen selected"
                          >
                            <input
                              checked=""
                              name="data-version"
                              type="radio"
                              value="1000"
                            />
                            <time
                              time="1970-01-02T10:17:36.789Z"
                            >
                              Jan. 2, 1970 – 10:17 a.m.
                            </time>
                          </label>
                        </li>
                      </ol>
                    </form>
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <form
                      class="notifications notifications-disabled"
                    >
                      <div
                        class="text"
                      >
                        <p
                          class="status"
                        >
                          <i
                            class="icon icon-no-notification"
                          />
                          js.WorkflowEditor.DataVersionModal.NotificationsForm.status.alertsOff
                        </p>
                        <p
                          class="description"
                        >
                          js.WorkflowEditor.DataVersionModal.NotificationsForm.description.turnAlertsOn
                        </p>
                      </div>
                      <div
                        class="options"
                      >
                        <label>
                          <input
                            name="notifications-enabled"
                            type="checkbox"
                          />
                          <span
                            class="action"
                          >
                            js.WorkflowEditor.DataVersionModal.NotificationsForm.options.turnOn
                          </span>
                        </label>
                      </div>
                    </form>
                    <div
                      class="actions"
                    >
                      <button
                        disabled=""
                        name="load"
                      >
                        Load
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </body>
        }
      >
        <div
          className="modal-backdrop"
        />
        <div
          className="modal data-versions-modal"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content"
            >
              <ModalHeader
                toggle={[MockFunction]}
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                  >
                    Data Versions
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    onClick={[MockFunction]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>
                </div>
              </ModalHeader>
              <ModalBody>
                <div
                  className="modal-body"
                >
                  <form
                    onCancel={[MockFunction]}
                    onSubmit={[Function]}
                  >
                    <ol>
                      <li
                        key="2000"
                      >
                        <FetchVersion
                          date={1970-01-03T17:09:27.890Z}
                          id="2000"
                          isSeen={false}
                          isSelected={false}
                          onSelect={[Function]}
                        >
                          <label
                            className="unseen"
                          >
                            <input
                              checked={false}
                              name="data-version"
                              onChange={[Function]}
                              type="radio"
                              value="2000"
                            />
                            <time
                              time="1970-01-03T17:09:27.890Z"
                            >
                              Jan. 3, 1970 – 5:09 p.m.
                            </time>
                          </label>
                        </FetchVersion>
                      </li>
                      <li
                        key="1000"
                      >
                        <FetchVersion
                          date={1970-01-02T10:17:36.789Z}
                          id="1000"
                          isSeen={true}
                          isSelected={true}
                          onSelect={[Function]}
                        >
                          <label
                            className="seen selected"
                          >
                            <input
                              checked={true}
                              name="data-version"
                              onChange={[Function]}
                              type="radio"
                              value="1000"
                            />
                            <time
                              time="1970-01-02T10:17:36.789Z"
                            >
                              Jan. 2, 1970 – 10:17 a.m.
                            </time>
                          </label>
                        </FetchVersion>
                      </li>
                    </ol>
                  </form>
                </div>
              </ModalBody>
              <ModalFooter>
                <div
                  className="modal-footer"
                >
                  <NotificationsForm
                    notificationsEnabled={false}
                    onSubmit={[Function]}
                  >
                    <form
                      className="notifications notifications-disabled"
                      onSubmit={[Function]}
                    >
                      <div
                        className="text"
                      >
                        <p
                          className="status"
                        >
                          <i
                            className="icon icon-no-notification"
                          />
                          <WithI18n
                            components={
                              Array [
                                <strong />,
                              ]
                            }
                            defaults="Alerts are <0>off</0>"
                            description="The tag adds emphasis"
                            id="js.WorkflowEditor.DataVersionModal.NotificationsForm.status.alertsOff"
                          >
                            <I18n
                              update={true}
                              withHash={true}
                            >
                              <Trans
                                components={
                                  Array [
                                    <strong />,
                                  ]
                                }
                                defaults="Alerts are <0>off</0>"
                                description="The tag adds emphasis"
                                id="js.WorkflowEditor.DataVersionModal.NotificationsForm.status.alertsOff"
                              >
                                <Render
                                  value="js.WorkflowEditor.DataVersionModal.NotificationsForm.status.alertsOff"
                                >
                                  js.WorkflowEditor.DataVersionModal.NotificationsForm.status.alertsOff
                                </Render>
                              </Trans>
                            </I18n>
                          </WithI18n>
                        </p>
                        <p
                          className="description"
                        >
                          <WithI18n
                            defaults="Turn alerts ON to receive an email if the output of this module changes"
                            id="js.WorkflowEditor.DataVersionModal.NotificationsForm.description.turnAlertsOn"
                          >
                            <I18n
                              update={true}
                              withHash={true}
                            >
                              <Trans
                                defaults="Turn alerts ON to receive an email if the output of this module changes"
                                id="js.WorkflowEditor.DataVersionModal.NotificationsForm.description.turnAlertsOn"
                              >
                                <Render
                                  value="js.WorkflowEditor.DataVersionModal.NotificationsForm.description.turnAlertsOn"
                                >
                                  js.WorkflowEditor.DataVersionModal.NotificationsForm.description.turnAlertsOn
                                </Render>
                              </Trans>
                            </I18n>
                          </WithI18n>
                        </p>
                      </div>
                      <div
                        className="options"
                      >
                        <label>
                          <input
                            checked={false}
                            name="notifications-enabled"
                            onChange={[Function]}
                            type="checkbox"
                          />
                          <span
                            className="action"
                          >
                            <WithI18n
                              defaults="Turn on"
                              id="js.WorkflowEditor.DataVersionModal.NotificationsForm.options.turnOn"
                            >
                              <I18n
                                update={true}
                                withHash={true}
                              >
                                <Trans
                                  defaults="Turn on"
                                  id="js.WorkflowEditor.DataVersionModal.NotificationsForm.options.turnOn"
                                >
                                  <Render
                                    value="js.WorkflowEditor.DataVersionModal.NotificationsForm.options.turnOn"
                                  >
                                    js.WorkflowEditor.DataVersionModal.NotificationsForm.options.turnOn
                                  </Render>
                                </Trans>
                              </I18n>
                            </WithI18n>
                          </span>
                        </label>
                      </div>
                    </form>
                  </NotificationsForm>
                  <div
                    className="actions"
                  >
                    <button
                      disabled={true}
                      name="load"
                      onClick={[Function]}
                    >
                      Load
                    </button>
                  </div>
                </div>
              </ModalFooter>
            </div>
          </div>
        </div>
      </Portal>
    </OpenModal>
  </Modal>
</DataVersionModal>
`;
